### PR TITLE
CLDR-17288 vagrant/ansible: fix for openliberty / mysql

### PR DESCRIPTION
**Pull request by srl295**
CLDR-17288

- this is just to get the build going again, CLDR-17279 will be for fixing docs etc.
- had to update the Ubuntu version (which we need in CLDR-16735) because of build issues, but should be forward/backward compatible.
- [x] This PR completes the ticket.

Discussion: https://github.com/orgs/community/discussions/73013

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by btangmu — DISMISSED** [review]

**Comment by srl295** [comment]
@btangmu I'm wondering if a better way to go about this would be to actually _provision and setup an azure machine_ as part of the testing. then we could run tests against it, and then shut it down.

**Comment by btangmu** [comment]
"actually provision and setup an azure machine as part of the testing" -- do you mean to integrate github with azure?

**Comment by srl295** [comment]
@btangmu 'testprovision' now succeeds and prints:

> `ERROR this usually fails but we are ignoring the failure. Try it manually`

Please review and then we can manually try `vagrant up` locally.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/XPathTable.java [hunk 1]
```diff
@@ -140,7 +140,6 @@ private void setupDB(Connection ourConn) throws SQLException {
             sql = null;
             s.close();
             s = null;
-            ourConn.commit();
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/XPathTable.java:143]: when called from SurveyMain, ourConn = getAConnection

but when called from TestXPathTable (twice) and TestSTFactory, ourConn = getDBConnection

seems like a bug either way; the callers need to be consistent?

**Review by btangmu — CHANGES_REQUESTED** [review]
LGTM except ourConn.commit() removal -- some callers use getDBConnection not getAConnection

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/XPathTable.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/XPathTable.java:143]: fixed

**Review by btangmu — APPROVED** [review]

**Pull request merged**